### DISCUSSION TITLE
Welcome to openEMS Discussions!

### DISCUSSION THREAD

**Discussion by thliebig** [thliebig/openEMS-Project#19] [Announcements]
## 👋 Welcome!
  Since the openEMS forum is down at the moment and it is unclear if it will return. I would like to encourage you to ask/discuss openEMS and all related issues here.
  * Ask questions about your models or how to use openEMS
  * Share ideas
  * Give Feedback

**Comment by alexpapad95** [comment]
Hello Thorsten. Is this our new forum? What about the previous one? Is it online?

**Reply by 0xCoto** [reply]
Seems like it for the time being:
> Yes that the forum is down is a real big issue and I have no idea if it will ever return or the university will just scrap everything after the hacker attack... I'm thinking to maybe give the github discussions a test...

For viewing old threads, I'd suggest using [Wayback Machine](https://web.archive.org/web/20221028194031/https://openems.de/forum/) for now.

**Reply by thliebig** [reply]
Yes it is for now. Maybe I'm able to at least recover a static version of the old forum. Is something like this possible from the Wayback Machine?

**Reply by 0xCoto** [reply]
I'm currently trying to fetch the whole thing using a tool called [`wayback-machine-downloader`](https://github.com/hartator/wayback-machine-downloader) - I'll leave it running overnight to grab everything (589/54681 pages so far), will keep you posted...

**Reply by thliebig** [reply]
Haha, me too I'm at 2330 already but it is really slow... lets see how this goes...

**Reply by 0xCoto** [reply]
Mine's still at 17660/54681 so I'm stopping it... :P hopefully yours is finished?

**Reply by thliebig** [reply]
It's not even close to that (6318/54681) because I did not want to let my machine run and consume power for only that... Luckily I can always restart and continue were I left off... this is going to take a while...

**Reply by Aspire-Design-Ltd** [reply]
Hi Thorsten, did the old forum download work? Last time I used openEMS I found the forum a great source of information.

**Comment by sajeedGIT** [comment]
Since I'm trying to use the python interface, can we install the python pre-module using pip?

**Reply by thliebig** [reply]
I'm not sure what you mean with "pre-module" (maybe pre-build module?). Well someday we/I should package it for pip at least for Windows, but you most likely will still need to download the windows build of openEMS since I do not want to include all this hundreds of *.dll in a pip package...   But with the current windows build it should still be fairly easy to [install](https://docs.openems.de/python/install.html#windows)?

**Comment by nasalmon** [comment]
Is there a Python code that runs openEMS that exploits software like Pytorch and TensorFlow to enable mathematical functions to be run on a GPU, to enable openEMS to run faster? Thank you, N

**Reply by thliebig** [reply]
No the python code is purely a cython wrapper around the C++ libraries. And openEMS does not use any fancy math functions for the simulation. It's just plain multiply and add. Not much more than that... The bottleneck is memory speed/bandwidth. Unfortunately even the GPU can not help here too much... I guess I really have to write a more detailed blog post about this to be able to go into more details...

**Comment by nasalmon** [comment]
I simply ask this as programs such a TensorFlow and Pytorch (modules which you can import to Python) have many prewritten algorithms to do common mathematical function (eg. FFT, convolution, matrix multiplication etc) where you just send the arrays to be processed to the GPU, without needing to write lots of extraneous code to make it happen. This strikes me as an opportunity to use some highly optimised code to process mathematical functions, that would otherwise have to be ported to the CPU. This might be an opportunity for openEMS? If so, I’d be interested to learn more about this, so I’ll look out for any future blogs in this area. Many thanks, N.

**Comment by Sarajevo67** [comment]
Hi guys,
any updates on the forum revival?

Looks like I got unlucky when I started using OpenEMS and got stuck with the Helix simulation just in time for the hosting to stop.
I didn't want to bother anyone, before browsing the forum, in search of already probably explained ambiguities.
I would kindly ask more experienced OpenEMS users for a couple of clarifications:

Is OpenEMS meshing and reliable Zin simulation of Helix antenna (in axial mode) made of thick wire (wrad 1 > mm) feasible?
If YES, what meshing approach, you recommend for such design?

**Reply by thliebig** [reply]
I'm indeed not sure if the Forum will ever return. It is for sure not on any priority for the university and my download from the web archives worked but the style is completely broken. So atm I'm not sure what to do about it.

As for your question. If you want Zin in relation to your wire thickness I would say no. If Zin just in relation to the antenna and e.g. diameter and winding numbers etc. I would say it should be okay. But in any case, I recommend to always only use 1D curves as wire.

**Comment by georgy-moshkin** [comment]
Is it possible to host downloaded archive somewhere so I can download it? I can try to create a html parser that will transform these pages into some usable format, e.g. an optimized html browsable website with a search which can be hosted somewhere, or zim-wiki or something like that. I am asking only because I have difficulties accessing webarchive without vpn, and my connection often drops. If not, can someone estimate amount of data which needs to be downloaded? Need some understanding how long it will take in my circumstances. Thanks!
> I'm indeed not sure if the Forum will ever return. It is for sure not on any priority for the university and my download from the web archives worked but the style is completely broken. So atm I'm not sure what to do about it.
> 
> As for your question. If you want Zin in relation to your wire thickness I would say no. If Zin just in relation to the antenna and e.g. diameter and winding numbers etc. I would say it should be okay. But in any case, I recommend to always only use 1D curves as wire.

**Reply by funderdown** [reply]
I agree with what you are saying. I am new to openEMS (not new to FDTD codes). 

If some type of getting started manual exist (or book on openEMS), it would be very helpful.

Failing that, is there a list of discussions so newbies can get up to speed on using openEMS?

**Comment by nasalmon** [comment]
Information in the forum is so important for a continued use of OpenEMS that it must continue. Without a forum, use of OpenEMS will just drop, as folks dont realise the amazing capability of this software. Without the forum, simulation will be done using other proprietary or other free-to-use packages. It's the kind of thing that needs public funding to be maintained, for the benefit of a wide community, enabling small business entities to do high-tech development.

**Reply by thliebig** [reply]
I agree that a forum is important, but I'm quite happy with this github discussions and all the actual discussions happening. Much more user feedback and interactions than in the old forum and most importantly I do not have to constantly waste my time with removing spam-bots....

**Reply by underdownf** [reply]
I agree with what you are saying. I am new to openEMS (not new to FDTD codes).

If some type of getting started manual exist (or book on openEMS), it would be very helpful.

Failing that, is there a list of discussions so newbies can get up to speed on using openEMS?

**Reply by funderdown** [reply]
I would like to start a discussion on learning how to use openEMS. A self learning book like: openEMS for Dummies. Something that is crystal clear, so we don't have to guess or try to modify a tutorial on how to use the software.

Maybe we can start by collecting articles that cover different aspects of using the software.

I would like to see this progress to a seminar course on Zoom where anyone can attend, to gain proficiency in setting up and running simulation.

Questions: Is this course something you all would be interested in building and taking?

Please let me know.

Cheers!

Frank

**Comment by nasalmon** [comment]
There’s certainly a huge amount of information already available in the comments from others. Anyone seriously interested in getting openEMS up and running can trawl through this to get what they want. However, I do think having some kind of introduction would help. I guess the questions are who takes editorial control and who contributes, as this needs to be something longer term and updated and refined if it is to be any good, not just a flash in the pan. OpenEMS is certainly a wonderful resource and it would be great to see something like a general introduction and guidance document for all aspects of the code and what it is capable of and how it can be possibly be made to run faster by whatever route.

 

Thank you, Neil

**Comment by underdownf** [comment]
From my research on openEMS, there is supposed to be a 1000 page manual for the software. Where is it?

Why are people having to try to convert tutorials into sketches they need. This is not effective way to learn openEMS.

Note: and just because you know how to program in Matlab, does not transfer into usable skills in openEMP. There is syntax in openEMS that does NOT exist in Matlab.

I turned to this software to have a better/ inexpensive alternative than HFSS. However, the cost in time, searching in the dark for appropriate syntax, not covered in the tutorials, is forcing me to consider other options, including writing FDTD software from stretch.

If you have access to a training manual, I would appreciate it if you would share it with me. Otherwise, I have wasted enough time with openEMS.

Cheers!

Frank

**Reply by LubomirJagos42** [reply]
Try look at this, it generates scripts from FreeCAD you have to define simulation using GUI similar to HFSS, this way you can at least make learning openEMS scripting faster:
  - https://www.youtube.com/watch?v=OR-79dCMdyo&t=305s
  - https://github.com/LubomirJagos42/FreeCAD-OpenEMS-Export/releases
  - https://github.com/LubomirJagos42/FreeCAD-OpenEMS-Export-Plugin-Documentation

**Comment by underdownf** [comment]
I am using Octave, to interface with openEMS.

There is a package in Octave, that is supposed to extend the geometry keywords available in Octave and hence openEMS.

The package is: geometry 4.1.0 (2024-03-19)

using pkg list, I can see the package is installed in Octave.  However, when reviewing the directory, I find the file geometry.m is not in the folder.  So, when I try to use this functionality (i.e.):

% Create a circle
circle = geometry.Circle(center=[0, 0, 0], radius=0.01);

This code does not work.  I get the error:

error: 'geometry' undefined near line 5, column 10
error: called from
    testEMS2 at line 5 column 8

Has anyone else run into this issue?  if so, how did you resolve it?

Note: I have updated to Octave version 9.2.0. I am using Windows 11 (not my first choice).

Cheers!

Frank
